core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -8782,6 +8782,12 @@ void testGroupExpressionEquivalenceParams() {
     sql("select stream * from orders left join products_temporal\n"
         + "for system_time as of CURRENT_TIMESTAMP\n"
         + "on orders.productid = products_temporal.productid").ok();
+
+    // verify left join of CTEs
```

Comment:
I think that this query is illegal according to the "standard" definition of SYSTEM_TIME.
You can make the method overridable, but is there even a definition of what it means to apply SYSTEM_TIME to the result of a query?
If you want it in here it should be applied to products_temporal.

Reply:
made it failing and overridable
>but is there even a definition of what it means to apply SYSTEM_TIME to the result of a query?

in Flink result of query in streaming mode is changelog. Changelog has timestamp, in this case we could jump at some position in the middle with SYSTEM_TIME